app/javascript/src/components/ChartCard/BarChartLegend.js

```diff
@@ -0,0 +1,21 @@
+import React from "react"
+import { useSelector } from "react-redux"
+import { getPlan, getDisplayForDiseaseId } from "../../config/selectors"
+
+const BarChartLegend = () => {
+  const plan = useSelector((state) => getPlan(state))
+  if (plan.disease_ids.length === 0) {
```

Comment:
This will fail with an error if `plan.disease_id` is anything other than an array. What about `null` or `undefined`? When a `Plan` has zero diseases applied will `disease_ids` still be an array in that case?

Reply:
the attributes method causes the to_json it to always return an array including when it's empty